internal/app/connectconformance/test_case_library.go

```diff
@@ -559,16 +559,6 @@ func populateExpectedResponse(testCase *conformancev1.TestCase) error {
 		return nil
 	}
 
-	// TODO - This is just a temporary constraint to protect against panics for now.
-	// Eventually, we want to be able to test client and bidi streams where there are no request messages.
-	// The potential plan is for server impls to produce (and the code below to expect) a single response
-	// message in this situation, where the response data value is some fixed string (such as "no response definition")
-	// and whose request info will still be present, but we expect it to indicate zero request messages.
-	if len(testCase.Request.RequestMessages) == 0 {
-		testCase.ExpectedResponse = &conformancev1.ClientResponseResult{}
-		return nil
-	}
```

Comment:
We now treat zero request messages as effectively the same as a request that has no response definition field. I did this so that I could add "no-requests" test cases, to test client and bidi streams where the request stream is empty. (However, I had to leave those new test cases commented out due to issues I ran into and haven't had a chance to finish debugging...)